core/src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -640,18 +667,19 @@ private IndexResult indexIntoLucene(Index index, long seqNo, long currentVersion
                  * `indexWriter.getTragicException()` will be null otherwise we have to rethrow and treat it as fatal or rather
                  * non-document failure
                  */
-                return new IndexResult(ex, currentVersion, index.seqNo());
+                return new IndexResult(ex, Versions.MATCH_ANY, index.seqNo());
```

Comment:
I don't have that information anymore here, and in the case of failure, we don't use it anyway. I'll comment.